### PR TITLE
redesign red screens layout

### PR DESCRIPTION
**Pull request by Chleba**
- add new informations into a red screens
- add new icon for red screens
- update layout of red screens

### CODE REVIEW

File: src/common/bsod.cpp [hunk 1]
```diff
@@ -264,12 +264,13 @@ void draw_error_screen(const uint16_t error_code_short) {
 
         /// draw "Scan me" text
         // r=1 c=34
-        static const char *scan_me_text = N_("Scan me for details");
-        render_text_align(Rect16(0, 142, display::GetW(), display::GetH() - 142), _(scan_me_text), resource_font(IDR_FNT_SMALL), COLOR_RED_ALERT, COLOR_WHITE, padding_ui8(0, 0, 0, 0), ALIGN_HCENTER);
+        // static const char *scan_me_text = N_("Scan me for details");
```
Comment (DRracer) [inline src/common/bsod.cpp:267]: please remove this comment, gettext may find this tagged text text

File: src/common/bsod.cpp [hunk 2]
```diff
@@ -264,12 +264,13 @@ void draw_error_screen(const uint16_t error_code_short) {
...
+        // render_text_align(Rect16(0, 142, display::GetW(), display::GetH() - 142), _(scan_me_text), resource_font(IDR_FNT_SMALL), COLOR_RED_ALERT, COLOR_WHITE, padding_ui8(0, 0, 0, 0), ALIGN_HCENTER);
 
         /// draw "Scan me" arrow
         /// FIXME arrow overlaps with QR code (bad PNG)
```
Comment (DRracer) [inline src/common/bsod.cpp:271]: also this FIXME is no longer valid

File: src/common/bsod.cpp [hunk 3]
```diff
@@ -301,7 +302,32 @@ void draw_error_screen(const uint16_t error_code_short) {
...
+
+        /// draw footer information
+        /// fw version, hash, [apendix], [fw signed]
+        /// fw version
+        char fw_version[32];
+        snprintf(fw_version, sizeof(fw_version), "%s%s", project_version, project_version_suffix_short);
+        render_text_align(Rect16(6, 290, 80, 10), string_view_utf8::MakeRAM((const uint8_t *)fw_version), resource_font(IDR_FNT_SMALL), COLOR_RED_ALERT, COLOR_WHITE, padding_ui8(0, 0, 0, 0), ALIGN_HCENTER);
+        /// hash
+        if (!qr_privacy) {
+            char p_code[] = {};
```
Comment (DRracer) [inline src/common/bsod.cpp:315]: this is zero sized array, this is definitely not correct

File: src/common/bsod.cpp [hunk 4]
```diff
@@ -301,7 +302,32 @@ void draw_error_screen(const uint16_t error_code_short) {
...
+            printerCode(p_code);
+            render_text_align(Rect16(98, 290, 64, 10), string_view_utf8::MakeRAM((const uint8_t *)p_code), resource_font(IDR_FNT_SMALL), COLOR_RED_ALERT, COLOR_WHITE, padding_ui8(0, 0, 0, 0), ALIGN_HCENTER);
+        }
+        /// [apendix, fw signed]
+        /// TODO: fw signed is not available ATM
+        /// signed fw
+        if (0) {
+            char signed_fw_str[4] = "[S]";
```
Comment (DRracer) [inline src/common/bsod.cpp:323]: static const, it doesn't have to be in RAM ... and use MakeCPUFLASH then (even though it is the same for now)

**Review by DRracer — CHANGES_REQUESTED** [review]
we may also remove the arrow PNG
and please add the changed POT file (remove "scan me")

File: src/common/bsod.cpp [hunk 5]
```diff
@@ -301,7 +295,32 @@ void draw_error_screen(const uint16_t error_code_short) {
         /// draw short URL
         error_url_short(qr_text, sizeof(qr_text), error_code);
         // this MakeRAM is safe - qr_text is a local buffer on stack
-        render_text_align(Rect16(0, 293, display::GetW(), display::GetH() - 293), string_view_utf8::MakeRAM((const uint8_t *)qr_text), resource_font(IDR_FNT_SMALL), COLOR_RED_ALERT, COLOR_WHITE, padding_ui8(0, 0, 0, 0), ALIGN_HCENTER);
+        render_text_align(Rect16(0, 255, display::GetW(), display::GetH() - 255), string_view_utf8::MakeRAM((const uint8_t *)qr_text), resource_font(IDR_FNT_SMALL), COLOR_RED_ALERT, COLOR_WHITE, padding_ui8(0, 0, 0, 0), ALIGN_HCENTER);
+
+        /// draw footer information
+        /// fw version, hash, [apendix], [fw signed]
+        /// fw version
+        char fw_version[32];
```
Comment (DRracer) [inline src/common/bsod.cpp:303]: let's limit this to the total number of really printable characters - abuse snprintf to clamp the string if too long

**Review by RadekVana — APPROVED** [review]

**Pull request merged**